src/SourceLocator/SourceStubber/PhpStormStubsSourceStubber.php

```diff
@@ -492,6 +494,11 @@ private function replaceExtendsOrImplementsByPhpVersion(string $className, array
                     $modifiedNames[] = new Node\Name\FullyQualified(Traversable::class);
                     continue;
                 }
+            } elseif ($className === SplObjectStorage::class) {
+                if ($name === SeekableIterator::class && $this->phpVersion < 80400) {
+                    $modifiedNames[] = new Node\Name\FullyQualified(Iterator::class);
+                    continue;
+                }
```

Comment:
I'd probably expect a test expansion here, or to rebase this on a pipeline that runs against 8.4

Reply:
Existing tests will fail as soon as PHPStorm make a new release, but currently there is none and it's on master
https://github.com/JetBrains/phpstorm-stubs/compare/v2024.2...master

Either I have to wait a new release,

Either I can:
- Bump phpstorm stubs to `dev-master` to show you the failing test
- Then remove the bump and keep `v2024.2` tag to merge this PR as an anticipated fix.

Your choice.

Reply:
I mostly don't want to leave a trail of unverified changes "for the future", heh: while the work you've done is great, it does indeed pile up at the risk of diverging with upstream stubs later on :|

Reply:
s/unverified/unverified by automation - you did do your due diligence :-)